test/database/upload_test.go

```diff
@@ -240,3 +239,41 @@ func TestUnpinUploads(t *testing.T) {
 			expectedUploadBandwidth, expectedUploadBandwidth/skynet.MB, stats.BandwidthUploads, stats.BandwidthUploads/skynet.MB)
 	}
 }
+
+// TestUploadCreateAnon ensures that UploadCreate can create anonymous uploads.
+func TestUploadCreateAnon(t *testing.T) {
+	ctx := context.Background()
+	dbName := test.DBNameForTest(t.Name())
+	db, err := database.NewCustomDB(ctx, dbName, test.DBTestCredentials(), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	sl := test.RandomSkylink()
+	skylink, err := db.Skylink(ctx, sl)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// Register an anonymous upload.
+	ip := "1.0.2.233"
+	up, err := db.UploadCreate(ctx, nil, ip, *skylink)
```

Comment:
kind of odd the return values for `UploadCreate` and `DownloadCreate` are not identical